parametric/test_headers_precedence.py

```diff
@@ -0,0 +1,544 @@
+from typing import Any
+
+import pytest
+
+from parametric.spec.tracecontext import get_tracecontext
+from parametric.utils.headers import make_single_request_and_get_inject_headers
+
+parametrize = pytest.mark.parametrize
+
+
+def enable_tracecontext() -> Any:
+    env = {
+        "DD_TRACE_PROPAGATION_STYLE": "tracecontext",
+    }
+    return parametrize("library_env", [env])
+
+
+def enable_datadog() -> Any:
+    env = {
+        "DD_TRACE_PROPAGATION_STYLE": "Datadog",
+    }
+    return parametrize("library_env", [env])
+
+
+def enable_datadog_tracecontext() -> Any:
+    env = {
+        "DD_TRACE_PROPAGATION_STYLE": "Datadog,tracecontext",
+    }
+    return parametrize("library_env", [env])
+
+
+@pytest.mark.skip_library("golang", "not implemented")
+@pytest.mark.skip_library("nodejs", "TODO: remove when https://github.com/DataDog/dd-trace-js/pull/2477 lands")
+def test_headers_precedence_propagationstyle_default(test_agent, test_library):
+    with test_library:
+        # 1) No headers
+        headers1 = make_single_request_and_get_inject_headers(test_library, [])
+
+        # 2) Only tracecontext headers
+        headers2 = make_single_request_and_get_inject_headers(
+            test_library, [["traceparent", "00-12345678901234567890123456789012-1234567890123456-01"],]
+        )
+
+        # 3) Only tracecontext headers, includes existing tracestate
+        headers3 = make_single_request_and_get_inject_headers(
+            test_library,
+            [["traceparent", "00-12345678901234567890123456789012-1234567890123456-01"], ["tracestate", "foo=1"],],
+        )
+
+        # 4) Both tracecontext and Datadog headers
+        headers4 = make_single_request_and_get_inject_headers(
+            test_library,
+            [
+                ["traceparent", "00-12345678901234567890123456789012-1234567890123456-01"],
+                ["tracestate", "foo=1"],
+                ["x-datadog-trace-id", "123456789"],
+                ["x-datadog-parent-id", "987654321"],
+                ["x-datadog-sampling-priority", "-2"],
+            ],
+        )
+
+        # 5) Only Datadog headers
+        headers5 = make_single_request_and_get_inject_headers(
+            test_library,
+            [
+                ["x-datadog-trace-id", "123456789"],
+                ["x-datadog-parent-id", "987654321"],
+                ["x-datadog-sampling-priority", "-2"],
```

Comment:
@zacharycmontoya As far as I know, the only [valid values](https://github.com/Datadog/architecture/blob/master/rfcs/apm/integrations/priority-sampling/rfc.md#sampling-values-and-propagation) for sampling priority are `-1`, `0`, `1` and `2`. Java enforces this and all these tests fail because of that. Would it be ok if I opened a PR with these set to `-1` instead?

Reply:
Yes that sounds good 👍🏼

Reply:
I think we should leave it as is. It seems that the tracers should indeed accept other incoming sampling priorities, and just pass them through. I'll make Java behave the same way.